src/Adyen/Service/AbstractResource.php

```diff
@@ -100,6 +101,88 @@ public function request($params, $requestOptions = null)
         return $curlClient->requestJson($this->service, $this->endpoint, $params, $requestOptions);
     }
 
+    /**
+     * Do the request to the Http Client
+     *
+     * @param $params
+     * @param null $requestOptions
+     * @return mixed
+     * @throws AdyenException
+     */
+    public function requestPatch($params, $requestOptions = null)
+    {
+        // convert to PHP Array if type is inputType is json
+        if ($this->service->getClient()->getConfig()->getInputType() == 'json') {
+            $params = json_decode($params, true);
+            if ($params === null && json_last_error() !== JSON_ERROR_NONE) {
+                $msg = 'The parameters in the request expect valid JSON but JSON error code found: ' .
+                    json_last_error();
+                $this->service->getClient()->getLogger()->error($msg);
```

Comment:
Dear @julian-van-drielen, thank you for your valuable contribution to support AfP in the Adyen PHP API Library.

There is an ongoing discussion to exclude all of the logging functionality in the library itself. I will discuss it with my colleagues after that logging can safely be deleted from here.